Decisions.Microsoft365.Exchange/GraphRest.cs

```diff
@@ -40,32 +40,33 @@ public static string Get(string urlExtension)
             }
         }
         
-        public static HttpResponseMessage HttpResponsePatch(string urlExtension, HttpContent content)
+        public static HttpResponseMessage HttpResponsePatch(ExchangeSettings? settingsOverride, string urlExtension, HttpContent content)
         {
-            return SendHttpRequest(urlExtension, content, HttpMethod.Patch);
+            return SendHttpRequest(settingsOverride, urlExtension, content, HttpMethod.Patch);
         }
 
-        public static string Patch(string urlExtension, HttpContent content)
+        public static string Patch(ExchangeSettings? settingsOverride, string urlExtension, HttpContent content)
         {
-            HttpResponseMessage response = HttpResponsePatch(urlExtension, content);
+            HttpResponseMessage response = HttpResponsePatch(settingsOverride, urlExtension, content);
             Task<string> resultTask = response.Content.ReadAsStringAsync();
             resultTask.Wait();
 
             return resultTask.Result;
         }
         
-        public static HttpResponseMessage Delete(string urlExtension)
+        public static HttpResponseMessage Delete(ExchangeSettings? settingsOverride, string urlExtension)
         {
-            return SendHttpRequest(urlExtension, null, HttpMethod.Delete);
+            return SendHttpRequest(settingsOverride, urlExtension, null, HttpMethod.Delete);
         }
         
-        private static HttpResponseMessage SendHttpRequest(string urlExtension, HttpContent? content,
-            HttpMethod httpMethod)
+        private static HttpResponseMessage SendHttpRequest(ExchangeSettings? settingsOverride, string urlExtension,
+            HttpContent? content, HttpMethod httpMethod)
         {
             ExchangeSettings settings = ModuleSettingsAccessor<ExchangeSettings>.GetSettings();
-            OAuthToken token = new ORM<OAuthToken>().Fetch(settings.TokenId);
-
-            string url = $"{settings.GraphUrl}{urlExtension}";
+            OAuthToken token = new ORM<OAuthToken>().Fetch(!string.IsNullOrEmpty(settingsOverride?.TokenId)
```

Comment:
For a little cleaner code, I might make this a GetSettings() method that will handle all the logic for finding whichever settings to use when and just returning a known good object.